File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.go
func findMatrix(nums []int) [][]int {
	answer := make([][]int, 0)
	hashMap := make(map[int]int)
	
	for _, v := range nums {
		hashMap[v]++;
	}
	
	for {
		if len(hashMap) == 0{
			return answer
		}
		arr := make([]int, 0)
		for k, v := range hashMap {
			arr = append(arr, k)
			value := v-1
			if value == 0 {
				delete(hashMap, k)
			} else {
				hashMap[k] = value
			}
		}
		answer = append(answer, arr)
	}

}

File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.java
    class Solution {
        public List<List<Integer>> findMatrix(int[] nums) {
            List<List<Integer>> answer = new ArrayList<>();

            HashMap<Integer, Integer> hashMap = new HashMap<>();
            for (int num : nums) {
                hashMap.compute(num, (k, v) -> v == null? 1 : v+1);
            }

            while (!hashMap.isEmpty()) {
                List<Integer> tempList = new ArrayList<>();
                Iterator<Map.Entry<Integer, Integer>> mapIterator = hashMap.entrySet().iterator();
                while (mapIterator.hasNext()) {
                    Map.Entry<Integer, Integer> entry = mapIterator.next();
                    tempList.add(entry.getKey());
                    if (entry.getValue() != null && entry.getValue() > 1) {
                        hashMap.put(entry.getKey(), entry.getValue()-1);
                    } else {
                        mapIterator.remove();
                    }
                }
                answer.add(tempList);
            }

            return answer;
        }
    }